src/inventorymanager.cpp

```diff
@@ -322,12 +322,20 @@ void IMoveAction::apply(InventoryManager *mgr, ServerActiveObject *player, IGame
 		return;
 	}
 
-	if ((u16)to_i > list_to->getSize()) {
-		infostream << "IMoveAction::apply(): FAIL: destination index out of bounds: "
-			<< "to_i=" << to_i
-			<< ", size=" << list_to->getSize() << std::endl;
+	if (from_i < 0 || list_from->getSize() <= (u32) from_i) {
+		infostream << "IMoveAction::apply(): FAIL: source index is not in range: "
```

Comment:
nitpicking and nobody will ever read this message but I think it sounds better
```suggestion
		infostream << "IMoveAction::apply(): FAIL: source index out of bounds: "
```